src/manager.rs

```diff
@@ -96,38 +88,12 @@ where
 
     /// `delete` should either complete the aggregate instance, along with all its associated events
     /// and transactional read side projections, or fail.
-    pub async fn delete(&self, aggregate_id: impl Into<Uuid> + Send) -> Result<(), AggregateManagerError<E>> {
-        self.event_store
-            .delete(aggregate_id.into())
-            .await
-            .map_err(AggregateManagerError::EventStore)
+    pub async fn delete(&self, aggregate_id: impl Into<Uuid> + Send) -> Result<(), E::Error> {
+        self.event_store.delete(aggregate_id.into()).await
     }
-}
 
-/// The Aggregate Manager may return errors of two types: `Aggregate` errors and `EventStore` errors.
-/// An `Aggregate` error is returned when a validation error occurs while handling a specific command
-/// within the underlying [`Aggregate`].
-/// On the other hand, an `EventStore` error can be generated during event insertion into the store,
-/// due to serialization issues, or as a result of a [`crate::handler::TransactionalEventHandler`] error.  
-pub enum AggregateManagerError<E>
-where
-    E: EventStore,
-{
-    Aggregate(<<E as EventStore>::Aggregate as Aggregate>::Error),
-    EventStore(<E as EventStore>::Error),
-}
-
-impl<E> Debug for AggregateManagerError<E>
-where
-    E: EventStore,
-{
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        match self {
-            AggregateManagerError::Aggregate(_) => {
-                let typename = std::any::type_name::<<<E as EventStore>::Aggregate as Aggregate>::Error>();
-                write!(f, "{}", typename)
-            }
-            AggregateManagerError::EventStore(err) => write!(f, "{:?}", err),
-        }
+    /// Returns the internal event store
+    pub fn event_store(&self) -> &E {
```

Comment:
Remind me why we need access to the EventStore?
Shared/immutable access, I might add - what are you going to do with it?

Reply:
Because if you want to keep an `AggregateManager` in the context and anyway get access to the event store (eg. in lira they want to persist events as they are since they consume the events from a queue and they wants to persist them. Honestly no clue why do they do this.. but..)

Reply:
Well is that a legitimate use case that we want to support? Freely persisting stuff?
Perhaps we do, need to double check and make it very explicit that `AggregateManager` is then only a convenience object and nothing more

Reply:
Since they are still able to create a PgStore from scratch i think that the answer is yes. 

What do you suggest in order to make more explicit the fact that is an helper and that's it?